src/sql/src/plan/statement/dml.rs

```diff
@@ -437,6 +437,14 @@ pub fn plan_explain_schema(
         mut statement,
     } = explain_schema;
 
+    // Force the sink's name to one that's guaranteed not to exist, by virtue of
+    // being a non-existent item in a schema under the system's control, so that
+    // `plan_create_sink` doesn't complain about the name already existing.
+    statement.name = Some(UnresolvedItemName::qualified(&[
+        "mz_catalog",
+        "mz_explain_schema",
+    ]));
```

Comment:
Note I think this change is required even with non-optional names. The existing code, AFAICT, will error if you happen to use the name of a sink that already exists, but given that `EXPLAIN ... SCHEMA` is not actually creating that sink, it doesn't seem like there's any need to error.